frontend/appflowy_flutter/packages/appflowy_popover/lib/src/popover.dart

```diff
@@ -78,11 +83,18 @@ class Popover extends StatefulWidget {
 
   final bool asBarrier;
 
+  /// The widget that will be used to trigger the popover.
+  ///
+  /// Why do we need this?
+  /// Because if the parent widget of the popover is GestureDetector,
+  ///  the conflict won't be resolve by using Listener, we want these two gestures exclusive.
+  final PopoverClickHandler clickHandler;
```

Comment:
cc @richardshiue

Reply:
Yeah this is an issue, our popover component needs more work. onTapdown isn't a very good UX.

The problem with using the gesture detector is that when you use it, the child button cannot specify an onTap or onPressed handler, not even `() {}` because it will take precedence. But setting it to null will disable the button's hover effects. (I was testing the `FlowyTextButton` in `SettingsLanguageView`) I think this can cause some confusion 

What if we change it to `onTapUp`?

Reply:
Why do we need to change it to `onTapUp`?

```dart
  Widget _buildClickHandler(Widget child, VoidCallback handler) {
    switch (widget.clickHandler) {
      case PopoverClickHandler.listener:
        return Listener(
          onPointerDown: (_) => handler(),
          child: child,
        );
      case PopoverClickHandler.gestureDetector:
        return GestureDetector(
          onTap: handler,
          child: child,
        );
    }
  }
```